azure_run/backup.py

```
@@ -1,175 +1,108 @@
+import json
+from azureml.core import Dataset
+
+_BACKUP_DATASTORE = "researcher_data"
+_BACKUP_ROOT_DIR  = "data-backup"
+
+def dataset(name, remote_path, version=None, overwrite_ok=False, post_validate_compare=True, post_validate_func=None):
+    global _BACKUP_DATASTORE, _BACKUP_ROOT_DIR
+    from . import log, datastore, dataset as load_dataset
+
+    # Validate
+    if type(name)!=str:        raise Exception(f"Invalid parameter 'name', expected type str.")
```

Ruff (0.8.2)

Use `is` and `is not` for type comparisons, or `isinstance()` for isinstance checks

(E721)